plugins/modules/cloud/smartos/vmadm.py

```diff
@@ -501,13 +560,9 @@ def create_payload(module, uuid):
 
     # Filter out the few options that are not valid VM properties.
     module_options = ['debug', 'force', 'state']
-    vmattrs = filter(lambda prop: prop not in module_options, p)
-
-    vmdef = {}
-
-    for attr in vmattrs:
-        if p[attr]:
-            vmdef[attr] = p[attr]
+    # @TODO make this a simple {} comprehension as soon as py2 is ditched
+    # @TODO {k: v for k, v in p.items() if k not in module_options}
+    vmdef = dict([(k, v) for k, v in p.items() if k not in module_options and v])
```

Comment:
I think for sake of readability, we should get rid of the `p` variable, since right now it is only used in precisely one place:
```suggestion
    vmdef = dict([(k, v) for k, v in mdule.params.items() if k not in module_options and v])
```
(Also remove its definition above)

Reply:
done, and I'm signing off for today. see you on the other side